CMakeLists.txt

```diff
@@ -123,25 +135,22 @@ target_link_libraries(
 file(WRITE empty.cpp)
 add_library(CouchbaseLiteC SHARED empty.cpp)
 target_include_directories(
-    CouchbaseLiteC
-    PUBLIC
-    include/cbl
-    ${PROJECT_BINARY_DIR}/generated_headers/public/cbl
+    CouchbaseLiteC PUBLIC 
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/vendor/couchbase-lite-core/vendor/fleece/API/>
+    $<INSTALL_INTERFACE:include/>
 )
+
 target_include_directories(
     CouchbaseLiteC
     PRIVATE
     src
     vendor/couchbase-lite-core/C
     ${PROJECT_BINARY_DIR}/generated_headers/private/cbl
+    ${PROJECT_BINARY_DIR}/generated_headers/public/cbl
```

Comment:
I think I don't totally understand how the target_include_directories actually works. When the header is public, why is it not put in the PUBLIC target?

Reply:
Oh you are right!  I just put it here without thinking because I consume it a different way when testing with the test server (when making the distributable I will collapse everything into include/cbl to make it easier for the consumer).